src/deployconfig.jl

```
@@ -843,6 +843,201 @@ authentication_method(::Buildkite) = Documenter.SSH
+    token_ok = env_nonempty("PROJECT_ACCESS_TOKEN")
+    auth_ok = owner_ok | token_ok
+    all_ok &= auth_ok
+
+    if token_ok
+        println(io, "- $(marker(token_ok)) ENV[\"PROJECT_ACCESS_TOKEN\"] exists and is non-empty")
+    end
+
+    if owner_ok
+        println(io, "- $(marker(owner_ok)) ENV[\"CI_REPO_OWNER\"] exists and is non-empty")
+    end
+
+    print(io, "Deploying: $(marker(all_ok))")
+    @info String(take!(io))
+    if build_type === :devbranch && !branch_ok && devbranch == "master" && cfg.woodpecker_ref == "refs/heads/main"
+        @warn """
+        Possible deploydocs() misconfiguration: main vs master. Current branch (from \$CI_COMMIT_REF) is "main". 
+        """
+    end
+
+    if all_ok
+        return DeployDecision(; all_ok=true,
+            branch=deploy_branch,
+            is_preview=is_preview,
+            repo=deploy_repo,
+            subfolder=subfolder
+        )
+    else
+        return DeployDecision(; all_ok=false)
+    end
+end
+
+authentication_method(::Woodpecker) = Documenter.HTTPS
+function authenticated_repo_url(cfg::Woodpecker)
+    if !isnothing(get(ENV, "FORGE_URL", nothing))
+        return "https://$(ENV["CI_REPO_OWNER"]):$(ENV["PROJECT_ACCESS_TOKEN"])@$(ENV["FORGE_URL"])/$(cfg.woodpecker_repo).git"
+    else
+        # Regex will return the root of the URL e.g. https://github.com/JuliaDocs/Documenter.jl → github.com
+        # `CI_REPO_LINK` is an env var which returns a repository link e.g. https://github.com/JuliaDocs/Documenter.jl
+        m = match(r"https?:\/\/(?:.*\.)*(.+\..+?)\/", get(ENV, "CI_REPO_LINK", ""))      
+        FORGE_URL=m.captures[1]
+        if !isempty(FORGE_URL)
+            return "https://$(ENV["CI_REPO_OWNER"]):$(ENV["PROJECT_ACCESS_TOKEN"])@$(FORGE_URL)/$(cfg.woodpecker_repo).git"
+        end
+    end
+  # Otherwise, raise an error
+    error("No repository URL or link found. Please check link or your `FORGE_URL` environment-variable again.")
```

I don't think `authenticated_repo_url` should raise an error. If the deployment is broken, it should be caught by `deploy_folder`. I would suggest extracting the necessary variables in the `Woopecker` constructor, checking them in `deploy_folder`, and just constructing the string here.

---

a `@warn` should be fine i guess

---

I don't think so. What would `authenticated_repo_url` return in that case? It should always return a sane value the `deplodocs` can use.